crates/sui-graphql-rpc/src/types/transaction_block_effects.rs

```diff
@@ -105,19 +115,76 @@ impl TransactionBlockEffects {
+    async fn errors(&self, ctx: &Context<'_>) -> Result<Option<String>> {
+        let mut status = self.native().status().clone();
+        let resolver: &PackageResolver = ctx.data_unchecked();
+
+        self.resolve_native_status_impl(resolver, &mut status)
+            .await?;
+
+        match status {
+            NativeExecutionStatus::Success => Ok(None),
 
             NativeExecutionStatus::Failure {
                 error,
                 command: None,
-            } => Some(error.to_string()),
+            } => Ok(Some(error.to_string())),
 
             NativeExecutionStatus::Failure {
                 error,
                 command: Some(command),
             } => {
+                let error = 'error: {
+                    let ExecutionFailureStatus::MoveAbort(loc, code) = &error else {
+                        break 'error error.to_string();
+                    };
+                    let fname_string = if let Some(fname) = &loc.function_name {
+                        format!("::{}'", fname)
+                    } else {
+                        "'".to_string()
+                    };
+
+                    let Some(CleverError {
+                        module_id,
+                        source_line_number,
+                        error_info,
+                    }) = resolver
+                        .resolve_clever_error(loc.module.clone(), *code)
+                        .await
+                    else {
+                        break 'error error.to_string();
+                    };
+
+                    match error_info {
+                        ConstantErrorInfo::Rendered {
+                            error_identifier,
+                            error_constant,
+                        } => {
+                            format!(
+                                "from '{}{fname_string} (line {source_line_number}), '{error_identifier}': {error_constant}",
+                                module_id.to_canonical_display(true)
+                            )
+                        }
+                        ConstantErrorInfo::Unrendered {
+                            error_identifier,
+                            error_bytes,
+                        } => {
+                            let const_str = FBase64::encode(error_bytes);
+                            format!(
+                                "from '{}{fname_string} (line {source_line_number}), '{error_identifier}': {const_str}",
+                                module_id.to_canonical_display(true)
+                            )
+                        }
+                        ConstantErrorInfo::None => {
+                            format!(
+                                "from '{}{fname_string} (line {source_line_number})",
+                                module_id.to_canonical_display(true)
+                            )
+                        }
+                    }
```

Comment:
nit (optional): Might be neater to `write!` to a string?

```rust
let error = 'error: {
    let ExecutionFailureStatus::MoveAbort(loc, code) = &error else {
        break 'error error.to_string();
    };

    let Some(CleverError {
        module_id,
        source_line_number,
        error_info,
    }) = resolver
        .resolve_clever_error(loc.module.clone(), *code)
        .await
    else {
        break 'error error.to_string();
    };

    let mut output = format!("from '{}", module_id.to_canonical_display(true));

    let _ = if let Some(fname) = &loc.function_name {
        write!(output, "::{fname}'")
    } else {
        write!(output, "'")
    }

    let _ = write!(output, " (line {source_line_number})");
    
    let _ = match error_info {
        ConstantErrorInfo::Rendered { 
            error_identifier,
            error_constant,
        } => write!(output, ", '{error_identifier}': {error_constant}"),

        ConstantErrorInfo::Unrendered {
            error_identifier,
            error_bytes,
        } => write!(output, ", '{error_identifier}': {}", FBase64::encode(error_bytes)),

        ConstantErrorInfo::None => { /* nop */ }
    };

    output
}
```

Reply:
I went to do this, but as I did I found it more confusing (at least to me), so going to keep the current `format` based way.